test/semantics/call05.f90

```diff
@@ -58,60 +62,72 @@ subroutine sup(x)
   subroutine sua(x)
     class(*), allocatable :: x(:)
   end subroutine
-  subroutine sdmp(x)
+  subroutine samp(x)
     type(pdt(*)), pointer :: x(:)
   end subroutine
+  subroutine sama(x)
+    type(pdt(*)), allocatable :: x(:)
+  end subroutine
+  subroutine sdmp(x)
+    type(pdt(:)), pointer :: x(:)
+  end subroutine
   subroutine sdma(x)
-    type(pdt(*)), pointer :: x(:)
+    type(pdt(:)), allocatable :: x(:)
   end subroutine
   subroutine snmp(x)
     type(pdt(1)), pointer :: x(:)
   end subroutine
   subroutine snma(x)
-    type(pdt(1)), pointer :: x(:)
+    type(pdt(1)), allocatable :: x(:)
   end subroutine
 
   subroutine test
     call smp(mp) ! ok
     call sma(ma) ! ok
     call spp(pp) ! ok
     call spa(pa) ! ok
-    !ERROR: If a dummy or effective argument is polymorphic, both must be so
+    !ERROR: If a POINTER or ALLOCATABLE dummy or actual argument is polymorphic, both must be so
     call smp(pp)
-    !ERROR: If a dummy or effective argument is polymorphic, both must be so
-    call sma(pp)
-    !ERROR: If a dummy or effective argument is polymorphic, both must be so
+    !ERROR: If a POINTER or ALLOCATABLE dummy or actual argument is polymorphic, both must be so
+    call sma(pa)
+    !ERROR: If a POINTER or ALLOCATABLE dummy or actual argument is polymorphic, both must be so
     call spp(mp)
-    !ERROR: If a dummy or effective argument is polymorphic, both must be so
-    call spa(mp)
-    !ERROR: If a dummy or effective argument is unlimited polymorphic, both must be so
+    !ERROR: If a POINTER or ALLOCATABLE dummy or actual argument is polymorphic, both must be so
+    call spa(ma)
+    !ERROR: If a POINTER or ALLOCATABLE dummy or actual argument is unlimited polymorphic, both must be so
     call sup(pp)
-    !ERROR: If a dummy or effective argument is unlimited polymorphic, both must be so
+    !ERROR: If a POINTER or ALLOCATABLE dummy or actual argument is unlimited polymorphic, both must be so
     call sua(pa)
-    !ERROR: If a dummy or effective argument is unlimited polymorphic, both must be so
+    !ERROR: actual argument type 'CLASS(*)' is not compatible with dummy argument type 't'
```

Comment:
If you put a "The" into the message string in lib/evaluate/characteristics.cc, ```IsComptibleWith()```, we could guarantee that the error message would consistently be capitalized.

Reply:
Yes, but that would not be amenable to translations into languages with fewer (Russian) or more (nearly all other Indo-European) than one form of their definite articles; I'm not sure it's worth doing.